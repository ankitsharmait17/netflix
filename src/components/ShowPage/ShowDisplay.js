import React from "react";
import { Container, Row, Col, Card, CardImg } from "reactstrap";
import { Link } from "react-router-dom";

const ShowDisplay = ({ showData }) => {
  return (
    <Container style={{ color: "white", marginTop: "20px" }}>
      <Row>
        <Link to="/" className="btn btn-secondary">
          Back
        </Link>
      </Row>
      <br />
      <Row>
        <Col lg="8">
          <h1>{showData.title}</h1>
        </Col>
        <Col lg="4">
          Rating:
          <span style={{ color: "yellow", fontSize: "40px" }}>&#9733;</span>
          <span style={{ fontSize: "40px" }}>{showData.rating}</span>/10
        </Col>
      </Row>
      <Row>
        <Col lg="1">({showData.year})</Col>
      </Row>
      <Row>
        <Col lg="3">
          <Card>
            <CardImg
              height="100%"
              src={`${window.location.origin}/images/posters/${showData.poster}`}
              alt={showData.title}
            ></CardImg>
          </Card>
        </Col>
        <Col lg="9" style={{ marginTop: "10px", borderRadius: "10px" }}>
          <iframe
            title={showData.title}
            src={`https://www.youtube.com/embed/${showData.trailer}`}
            width="100%"
            height="102%"
            frameBorder="10px"
            allowFullScreen="yes"
          ></iframe>
        </Col>
      </Row>
      <br />
      <Row>
        <Col lg="12">
          <b>Synopsis:</b>
          <p>{showData.description}</p>
        </Col>
      </Row>
    </Container>
  );
};

export default ShowDisplay;
