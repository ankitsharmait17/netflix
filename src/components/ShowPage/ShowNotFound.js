import React from "react";
import { Container, Row, Col } from "reactstrap";
import { Link } from "react-router-dom";

const ShowNotFound = () => {
  return (
    <Container style={{ color: "white", marginTop: "20px" }}>
      <Row>
        <Link to="/" className="btn btn-secondary">
          Back
        </Link>
      </Row>
      <br />
      <Row>
        <Col lg="12" className="text-center">
          <h1>Show not found.</h1>
        </Col>
      </Row>
    </Container>
  );
};

export default ShowNotFound;
