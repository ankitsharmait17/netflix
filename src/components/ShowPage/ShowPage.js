import React, { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import { Container, Row, Col, Card, CardImg } from "reactstrap";
import * as imdbAPI from "../../api/imdbAPI";
import data from "../../data.json";
import Spinner from "../Common/Spinner";
import PropTypes from "prop-types";

const ShowPage = ({ match }) => {
  const [showData, setshowData] = useState({
    title: "",
    rating: "",
    year: "",
    poster: "",
    trailer: "",
    description: ""
  });
  const [isLoading, setIsLoading] = useState(false);
  const getData = async () => {
    setIsLoading(true);
    const id = match.params.id;
    const show = data.shows.find(element => element.imdbID === id);
    const result = await imdbAPI.getImdbRating(id);
    setshowData({
      title: show.title,
      description: show.description,
      rating: result.imdbRating,
      year: show.year,
      poster: show.poster,
      trailer: show.trailer
    });
    setIsLoading(false);
  };
  useEffect(() => {
    getData();
  }, []);
  return isLoading ? (
    <Spinner />
  ) : (
    <Container style={{ color: "white", marginTop: "20px" }}>
      <Row>
        <Link to="/" className="btn btn-secondary">
          Back
        </Link>
      </Row>
      <br />
      <Row>
        <Col lg="8">
          <h1>{showData.title}</h1>
        </Col>
        <Col lg="4">
          <span style={{ color: "yellow", fontSize: "40px" }}>&#9733;</span>
          <span style={{ fontSize: "40px" }}>{showData.rating}</span>/10
        </Col>
      </Row>
      <Row>
        <Col lg="1">({showData.year})</Col>
      </Row>
      <Row>
        <Col lg="3">
          <Card>
            <CardImg
              height="100%"
              src={`${window.location.origin}/images/posters/${showData.poster}`}
              alt={showData.title}
            ></CardImg>
          </Card>
        </Col>
        <Col lg="9" style={{ marginTop: "10px", borderRadius: "10px" }}>
          <iframe
            title={showData.title}
            src={`https://www.youtube.com/embed/${showData.trailer}`}
            width="100%"
            height="102%"
            frameBorder="10px"
            allowFullScreen="yes"
          ></iframe>
        </Col>
      </Row>
      <br />
      <Row>
        <Col lg="12">
          <b>Synopsis:</b>
          <p>{showData.description}</p>
        </Col>
      </Row>
    </Container>
  );
};

ShowPage.propTypes = {
  match: PropTypes.object
};

export default ShowPage;
