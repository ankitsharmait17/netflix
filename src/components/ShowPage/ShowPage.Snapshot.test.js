import React from "react";
import ShowPage from "./ShowPage";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

it("passing correct imdbID,should render,Show display component", () => {
  const tree = renderer.create(
    <BrowserRouter>
      <ShowPage match={{ params: { id: "tt1856010" } }} />
    </BrowserRouter>
  );
  expect(tree).toMatchSnapshot();
});

it("passing incorrect imdbID,should render,ShowNotFound component", () => {
  const tree = renderer.create(
    <BrowserRouter>
      <ShowPage match={{ params: { id: "2" } }} />
    </BrowserRouter>
  );
  expect(tree).toMatchSnapshot();
});
