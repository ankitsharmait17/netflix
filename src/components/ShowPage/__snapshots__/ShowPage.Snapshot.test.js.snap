// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passing correct imdbID,should render,Show display component 1`] = `
<div
  className="container"
  style={
    Object {
      "color": "white",
      "marginTop": "20px",
    }
  }
>
  <div
    className="row"
  >
    <a
      className="btn btn-secondary"
      href="/"
      onClick={[Function]}
    >
      Back
    </a>
  </div>
  <br />
  <div
    className="row"
  >
    <div
      className="col-lg-8"
    >
      <h1>
        
      </h1>
    </div>
    <div
      className="col-lg-4"
    >
      Rating:
      <span
        style={
          Object {
            "color": "yellow",
            "fontSize": "40px",
          }
        }
      >
        â˜…
      </span>
      <span
        style={
          Object {
            "fontSize": "40px",
          }
        }
      >
        
      </span>
      /10
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-lg-1"
    >
      (
      
      )
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-lg-3"
    >
      <div
        className="card"
      >
        <img
          alt=""
          className="card-img"
          height="100%"
          src="http://localhost/images/posters/"
        />
      </div>
    </div>
    <div
      className="col-lg-9"
      style={
        Object {
          "borderRadius": "10px",
          "marginTop": "10px",
        }
      }
    >
      <iframe
        allowFullScreen="yes"
        frameBorder="10px"
        height="102%"
        src="https://www.youtube.com/embed/"
        title=""
        width="100%"
      />
    </div>
  </div>
  <br />
  <div
    className="row"
  >
    <div
      className="col-lg-12"
    >
      <b>
        Synopsis:
      </b>
      <p>
        
      </p>
    </div>
  </div>
</div>
`;

exports[`passing incorrect imdbID,should render,ShowNotFound component 1`] = `
<div
  className="container"
  style={
    Object {
      "color": "white",
      "marginTop": "20px",
    }
  }
>
  <div
    className="row"
  >
    <a
      className="btn btn-secondary"
      href="/"
      onClick={[Function]}
    >
      Back
    </a>
  </div>
  <br />
  <div
    className="row"
  >
    <div
      className="text-center col-lg-12"
    >
      <h1>
        Show not found.
      </h1>
    </div>
  </div>
</div>
`;
